tycho-bundles/org.eclipse.tycho.p2.resolver.shared/src/main/java/org/eclipse/tycho/p2/resolver/facade/P2ResolverFactory.java

```diff
@@ -30,4 +33,6 @@
     public TargetPlatformFactory getTargetPlatformFactory();
 
     public P2Resolver createResolver(MavenLogger logger);
+
+    MavenDependencyDescriptor getDependencyDescriptor(ArtifactDescriptor artifactDescriptor);
```

Comment:
I think `resolve` is more correct than `get` here. Also we should document whether the output is nullable and in which case it is.